code/game/machinery/lightswitch.dm

```diff
@@ -182,3 +182,7 @@
 		light_connect = !light_connect
 	if("toggle_logic" in href_list)
 		logic_connect = !logic_connect
+
+/obj/machinery/light_switch/off/Initialize()
```

Comment:
I'd rather you have the `/obj/machinery/light_switch` check if the `on` var is set. If so, take that value. If not, use the existing logic.
You'd have to remove the default value of `on` for this. So all normal lights will follow the default logic. And this one will be off.

Also, a sidenote, calling `attack_hand` without a user actually doing it is bad practice as it can lead to a lot of weird things.

Reply:
Noted! For now, I replaced this switch with a normal switch, because I realized it makes no sense for this to start in an off state. Lights start in an off state anyhow without power!